dhis-2/dhis-services/dhis-service-setting/src/main/java/org/hisp/dhis/setting/SettingKey.java

```diff
@@ -163,7 +166,18 @@
     ANALYTICS_HIDE_DAILY_PERIODS( "keyHideDailyPeriods", Boolean.FALSE, Boolean.class ),
     ANALYTICS_HIDE_WEEKLY_PERIODS( "keyHideWeeklyPeriods", Boolean.FALSE, Boolean.class ),
     ANALYTICS_HIDE_MONTHLY_PERIODS( "keyHideMonthlyPeriods", Boolean.FALSE, Boolean.class ),
-    ANALYTICS_HIDE_BIMONTHLY_PERIODS( "keyHideBiMonthlyPeriods", Boolean.FALSE, Boolean.class );
+    ANALYTICS_HIDE_BIMONTHLY_PERIODS( "keyHideBiMonthlyPeriods", Boolean.FALSE, Boolean.class ),
+
+    /**
+     * The Analytics query caching factor. It's used as a factor to assist with the
+     * caching TTL calculation.
+     */
+    ANALYTICS_CACHE_FACTOR( "keyAnalyticsCacheFactor", 160, Integer.class ),
+
+    /**
+     * The cache type enabled for Analytics. It can be FIXED or PROGRESSIVE.
+     */
+    ANALYTICS_CACHE_TYPE( "keyAnalyticsCacheType", "FIXED", String.class );
```

Comment:
I suggest we change the value to be an Enum. We do actually support enums as system setting values as the value is Serializable. That way we make it type-safe.

Reply:
Done